codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.util.CharsetUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+public final class RedisMessageFactory {
+
+    private static final PredefinedString[] PREDEFINED_SIMPLE_STRINGS = {
+            new PredefinedString(RedisMessageType.SIMPLE_STRING, "OK"),
+            new PredefinedString(RedisMessageType.SIMPLE_STRING, "QUEUED")
+            // TODO add predefined simple strings for Redis responses
+    };
+
+    private static final PredefinedString[] PREDEFINED_ERRORS = {
+            new PredefinedString(RedisMessageType.ERROR, "ERR")
+            // TODO add predefined errors for Redis responses
+    };
+
+    private static final ConstantBulkStringRedisMessage NULL_BULK_STRING = new ConstantBulkStringRedisMessage(null);
```

Comment:
I'm not sure about `Unpooled..` because `RefCountedBulkStringRedisMessage` is `Unpooled..` too.